Microsoft.Alm.Authentication/Src/Git/Utilities.cs

```diff
@@ -122,7 +125,9 @@ internal static IEnumerable<Win32.ProcessEntry32> EnumerateParentProcesses()
                 }
             }
 
-            var processList = new LinkedList<Win32.ProcessEntry32>();
+            // Allocate an array of process entries, 32 should be more than enough to walk from here back to the sought parent entry,
```

Comment:
Could you not use a `HashSet<Win32.ProcessEntry32>` instead and then change the loop below to:
```csharp
while (processTable.TryGetValue(processId, out processEntry) // do we have the process entry for this PID?
    && processList.Add(processEntry))                        // true if we haven't added this one before
{
    processId = processEntry.ParentProcessId;
}
```

Seeing as, per your comment, we don't expect the depth to be more than 32 then using a `HashSet` of _usually_ < 32 values should be more than fast enough, without leading to a case where this fails for deep process hierarchies in some cases.

I can totally imagine some crazy popular tool/script/console out there that result in deep processes trees.

Reply:
seems expensive, and for what?

Reply:
(replying to myself).. but I guess the set has no order then... hmm..